protocol/app/prepare/prices/default_price_update_generator.go

```diff
@@ -0,0 +1,36 @@
+package prices
+
+import (
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	pricestypes "github.com/dydxprotocol/v4-chain/protocol/x/prices/types"
+)
+
+// PreparePricesKeeper defines the expected Prices keeper used for `PrepareProposal`.
+type PricesKeeper interface {
+	GetValidMarketPriceUpdates(ctx sdk.Context) *pricestypes.MsgUpdateMarketPrices
+}
+
+// DefaultPriceUpdateGenerator is the default implementation of the PriceUpdateGenerator interface.
+// This implementation retrieves the `MsgUpdateMarketPrices` from the `PricesKeeper`, i.e the default
+// behavior for dydx v4's PrepareProposalHandler.
+type DefaultPriceUpdateGenerator struct {
+	// pk is an adapter for the
+	pk PricesKeeper
+}
+
+// NewDefaultPriceUpdateGenerator returns a new DefaultPriceUpdateGenerator.
+func NewDefaultPriceUpdateGenerator(keeper PricesKeeper) *DefaultPriceUpdateGenerator {
```

Comment:
when is this ever used? Shouldn't this replace SlinkyPriceUpdateGenerator if we set a flag? I dont see this being called anywhere

Reply:
Updated--current version can run either VE-based prepare/process or existing proposer based oracle logic. And we can switch the sidecar/pricefeed daemon on and off as well.